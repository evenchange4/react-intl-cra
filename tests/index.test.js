/**
 * @jest-environment node
 *
 * @flow
 */

const { execSync } = require('child_process');
const extract = require('../src/index');

it('should extract with source pattern', () => {
  const srcPattern = './tests/cra-project/src/**/*.js';
  // Fn
  expect(extract(srcPattern)).toMatchSnapshot();
  // CLI
  expect(
    execSync(`./bin/index.js '${srcPattern}'`).toString()
  ).toMatchSnapshot();
});

it('should extract with one file', () => {
  const srcPattern = './tests/cra-project/src/Header.js';
  // Fn
  expect(extract(srcPattern)).toMatchSnapshot();
  // CLI
  expect(
    execSync(`./bin/index.js '${srcPattern}'`).toString()
  ).toMatchSnapshot();
});

it('should print --help', () => {
  expect(execSync(`./bin/index.js --help`).toString()).toMatchSnapshot();
});

it('should log error', () => {
  try {
    execSync(`./bin/index.js`);
  } catch (error) {
    expect(error).toMatchSnapshot();
  }
});
