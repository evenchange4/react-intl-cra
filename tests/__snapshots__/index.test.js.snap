// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should extract with one file 1`] = `
Array [
  Object {
    "defaultMessage": "React Intl CRA",
    "description": "Library name",
    "filepath": "./tests/cra-project/src/Header.js",
    "id": "Header.name",
  },
  Object {
    "defaultMessage": "Extract messages of \`Creact React App\` from the command line",
    "description": "Library description",
    "filepath": "./tests/cra-project/src/Header.js",
    "id": "Header.desc",
  },
]
`;

exports[`should extract with one file 2`] = `
"[
  {
    \\"id\\": \\"Header.name\\",
    \\"description\\": \\"Library name\\",
    \\"defaultMessage\\": \\"React Intl CRA\\",
    \\"filepath\\": \\"./tests/cra-project/src/Header.js\\"
  },
  {
    \\"id\\": \\"Header.desc\\",
    \\"description\\": \\"Library description\\",
    \\"defaultMessage\\": \\"Extract messages of \`Creact React App\` from the command line\\",
    \\"filepath\\": \\"./tests/cra-project/src/Header.js\\"
  }
]
"
`;

exports[`should extract with source pattern 1`] = `
Array [
  Object {
    "defaultMessage": "Hello {name}, you have {unreadCount, number} {unreadCount, plural, one {message} other {messages}}",
    "description": undefined,
    "filepath": "./tests/cra-project/src/App.js",
    "id": "App.message",
  },
  Object {
    "defaultMessage": "React Intl CRA",
    "description": "Library name",
    "filepath": "./tests/cra-project/src/Header.js",
    "id": "Header.name",
  },
  Object {
    "defaultMessage": "Extract messages of \`Creact React App\` from the command line",
    "description": "Library description",
    "filepath": "./tests/cra-project/src/Header.js",
    "id": "Header.desc",
  },
]
`;

exports[`should extract with source pattern 2`] = `
"[
  {
    \\"id\\": \\"App.message\\",
    \\"defaultMessage\\": \\"Hello {name}, you have {unreadCount, number} {unreadCount, plural, one {message} other {messages}}\\",
    \\"filepath\\": \\"./tests/cra-project/src/App.js\\"
  },
  {
    \\"id\\": \\"Header.name\\",
    \\"description\\": \\"Library name\\",
    \\"defaultMessage\\": \\"React Intl CRA\\",
    \\"filepath\\": \\"./tests/cra-project/src/Header.js\\"
  },
  {
    \\"id\\": \\"Header.desc\\",
    \\"description\\": \\"Library description\\",
    \\"defaultMessage\\": \\"Extract messages of \`Creact React App\` from the command line\\",
    \\"filepath\\": \\"./tests/cra-project/src/Header.js\\"
  }
]
"
`;

exports[`should log error 1`] = `
[Error: Command failed: ./bin/index.js
Missing <pattern>
]
`;

exports[`should print --help 1`] = `
"Usage: index.js <pattern> [options]

<pattern> Glob pattern to specify files.
          Needs to be surrounded with quotes to prevent shell globbing.
          Guide to globs: https://github.com/isaacs/node-glob

Options:
  -o, --out-file  Output into a single file                             [string]
  -h, --help      Show help                                            [boolean]
  -v, --version   Show version number                                  [boolean]

Examples:
  index.js 'src/App.js'                   One file.
  index.js 'src/**/*.js'                  Pattern to specify files
  index.js 'src/**/*.js' -o message.json  Output into a single file.

For more information go to https://github.com/evenchange4/react-intl-cra

"
`;
